cmd/snap-exec/main.go

```diff
@@ -137,6 +137,9 @@ func snapExecApp(snapApp, revision, command string, args []string) error {
 
 	// run the command
 	fullCmd := filepath.Join(app.Snap.MountDir(), cmd)
+	if command == "shell" {
+		fullCmd = "/bin/bash"
```

Comment:
Hm, this will run inside the confinement with ubuntu-core as the base. So if your shell happens to be `zsh` this won't work inside this environment.